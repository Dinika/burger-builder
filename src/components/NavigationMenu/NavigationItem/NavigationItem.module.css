.NavigationItem {
    margin: 10px 0;
    box-sizing: border-box;
    display: block;
    width: 100%;
}

.NavigationItem a {
    color: #333;
    text-decoration: none;
    width: 100%;
    box-sizing: border-box;
    display: block;
}

.NavigationItem a:hover,
.NavigationItem a:active,
.NavigationItem a.active {
    color: rgb(35, 141, 228);
}

@media (min-width: 600px) {
    .NavigationItem {
        margin: 0;
        display: flex;
        height: 100%;
        width: auto;
        align-items: center;
    }
    
    .NavigationItem a {
        color: white;
        height: 100%;
        padding: 16px 10px;
        border-bottom: 4px solid transparent;
    }
    
    .NavigationItem a:hover,
    .NavigationItem a:active,
    .NavigationItem a.active {
        background-color: #444;
        border-bottom: 4px solid #eee;
        color: whitesmoke;
    }
}

