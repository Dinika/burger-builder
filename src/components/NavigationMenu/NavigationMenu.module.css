.NavigationMenu {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
}

@media (min-width: 600px) {
    .NavigationMenu {
        flex-direction: row;
    }
}