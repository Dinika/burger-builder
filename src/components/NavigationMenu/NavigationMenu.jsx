import React from 'react';
import NavigationItem from './NavigationItem/NavigationItem';
import classes from './NavigationMenu.module.css';

const NavigationMenu = (props) => (
    <ul className={classes.NavigationMenu}>
        <NavigationItem link="/">BurgerBuilder</NavigationItem>
        <NavigationItem link="/orders">Orders</NavigationItem>
    </ul>
)

export default NavigationMenu;