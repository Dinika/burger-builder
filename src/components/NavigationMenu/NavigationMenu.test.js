import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import NavigationItem from './NavigationItem/NavigationItem';
import NavigationMenu from './NavigationMenu';

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('NavigationMenu', () => {
  it('renders with no props passed', () => {
    const wrapper = shallow(<NavigationMenu />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 2 NavigationLinks if not authenticated', () => {
    const wrapper = shallow(<NavigationMenu />);
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });
});
