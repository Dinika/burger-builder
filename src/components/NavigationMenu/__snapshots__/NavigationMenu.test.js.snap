// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu renders with no props passed 1`] = `
<ul
  className="NavigationMenu"
>
  <NavigationItem
    link="/"
  >
    BurgerBuilder
  </NavigationItem>
  <NavigationItem
    link="/login"
  >
    Login
  </NavigationItem>
</ul>
`;
